File: 2023 November POTD/Reverse a Doubly Linked List.java
public static Node reverseDLL(Node head)

{

    //Your code here

    Node curr = head, next = head;

    while(curr!=null){

        next = curr.next;

        curr.next = curr.prev;

        curr.prev = next;

        if(next==null)

            break;

        curr = next;

    }

    return curr;

}

File: 2025 September POTD/Reverse a Doubly Linked List.cpp
class Solution {
  public:
    Node *reverse(Node *head) {
        // code here
        Node* right = head;
        int cnt = 1;
        while(right != NULL && right -> next != NULL) right = right -> next, cnt++;
        Node* left = head;
        
        int l = 1, r = cnt;
        while(l < r){
            swap(left -> data, right -> data);
            left = left -> next;
            right = right -> prev;
            l++, r--;
        }
        return head;
    }
};
